browser-tests/childrenFeature.ts

```diff
@@ -0,0 +1,126 @@
+import { login } from './utils/login';
+import getDropdownOption from './utils/getDropdownOption';
+import {
+  route,
+  godchildrenProfilePage,
+  addChildModal,
+  selectChild,
+  deleteChild,
+} from './pages/godchildrenProfilePage';
+import {
+  childrenProfilePage,
+  editChildModal,
+} from './pages/childrenProfilePage';
+
+function buildAddChild() {
+  return {
+    birthDate: {
+      day: '02',
+      month: '02',
+      year: '2020',
+    },
+    city: 'Helsinki',
+    postalCode: '00000',
+    firstName: 'Gilly',
+    lastName: 'Girod',
+    relationship: 'Vanhempi',
+  };
+}
+
+function buildEditChild() {
+  return {
+    lastName: `Citron ${new Date().toLocaleDateString()}`,
+  };
+}
+
+const childName = /Hertta Citron .*/;
+
+fixture`Children feature`
+  .page(route())
+  .beforeEach(async (t) => {
+    await login(t);
+    // Wait in order to allow authentication to complete
+    await t.wait(2500); // 2s
+
+    t.ctx.addChild = buildAddChild();
+    t.ctx.editChild = buildEditChild();
+  })
+  .afterEach(async (t) => {
+    delete t.ctx.addChild;
+    delete t.ctx.editChild;
+  });
+
+test('As a guardian I want to see a list of my children and to be able to select one', async (t) => {
+  // The list displays the expected fields
+  await t.expect(godchildrenProfilePage.children.count).gte(2);
+  await selectChild(t, childName);
+  await t.expect(childrenProfilePage.childName.textContent).match(childName);
+});
+
+test('As a guardian I want to edit the details of my child', async (t) => {
+  const nextLastName = t.ctx.editChild.lastName;
+
+  // Select child to go to their details
+  await selectChild(t, childName);
+
+  // Open child edit modal
+  await t.click(childrenProfilePage.editChildProfileButton);
+
+  // Check that the modal is semantically correct
+  await t.expect(editChildModal.container.exists).ok();
+
+  await t
+    // Clear last name
+    .selectText(editChildModal.lastNameInput)
+    .pressKey('delete')
+    // Input new last name
+    .typeText(editChildModal.lastNameInput, nextLastName)
```

Comment:
Technically the `.pressKey('delete)` is not needed, while the text is selected the next typeText will override current text.

Reply:
You are right 👍 I changed this.